data_structures.ipynb

```diff
@@ -0,0 +1,579 @@
+    "in convenience. Items can be addressed with accessors:\n",
+    "\n",
+    "```python\n",
+    "# This creates an empty array. If we try to assign a value to any position, it will flag an error.\n",
+    "array = []\n",
+    "# Now let's play a little bit with the language to populate the array.\n",
+    "array = [0] * 5 # This creates a list of 5 elements, all of them 0.\n",
+    "print(f\"array={array}\")\n",
+    "# Once the elements are reserved in memory, then we can play with the accessors.\n",
+    "array[0] = 1\n",
+    "# The following line uses the accessor ('[', ']') to obtain the value of the first element.\n",
+    "print(f\"array[0]={array[0]}\")\n",
+    "```\n",
+    "\n",
+    "As you can notice, mutability is one of the features from this data structure that is relevant to highlight since it will allow us to modify the contained values once created.\n",
+    "\n",
+    "### Example\n",
+    "\n",
+    "Let's transform our first code snipet into a list to see how much the code is going to be reduced and reused."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "var_abc[97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122]\n",
+      "var_abc[97, 98, 99, 100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122]\n",
+      "var_ABC[65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90]\n",
+      "alpha_abc['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']\n",
+      "alpha_ABC['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z']\n",
+      "reversed alpha_ABC['Z', 'Y', 'X', 'W', 'V', 'U', 'T', 'S', 'R', 'Q', 'P', 'O', 'N', 'M', 'L', 'K', 'J', 'I', 'H', 'G', 'F', 'E', 'D', 'C', 'B', 'A']\n",
+      "sorted alpha_ABC['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z']\n"
+     ]
+    }
+   ],
+   "source": [
+    "var_abc = list()\n",
+    "\n",
+    "# let's populate the list\n",
+    "for idx in range(97, 123): # range works as the interval [97, 123[\n",
+    "    var_abc.append(idx)\n",
+    "\n",
+    "print(f\"var_abc{var_abc}\")\n",
+    "\n",
+    "# To uppercase\n",
+    "def to_upper(char):\n",
+    "    return char - 32\n",
+    "\n",
+    "# lists can also be defined with square brackets\n",
+    "var_ABC = []\n",
+    "\n",
+    "# let's traverse var_abc and populate var_ABC\n",
+    "for value in var_abc:\n",
+    "    value_uppercase = to_upper(value)\n",
+    "    var_ABC.append(value_uppercase)\n",
+    "\n",
```

Comment:
I think you should clearly highlight what is the portion of code being improved here as it looks that there is extra staff being done after using the to_upper function. Also, perhaps because of the '...' used in the previously snippet code this actually looks bigger not reduced.